### PR TITLE
THREESCALE-11006 Quick Start content- Main Page Part 2

### PR DESCRIPTION
**Pull request by lvillen**
**What this PR does / why we need it**:

Now that Quick starts are available, we want to make them more accessible. In order to achieve that, the design team proposed two different actions:

1. Adding a Quick starts button to the dashboard. 
Captura de pantalla 2024-03-18 a las 16 34 23

2. When users log in for the first time, they should receive a notification informing them that Quick starts are available and encouraging them to take a tour.
Captura de pantalla 2024-03-18 a las 16 41 19

**Which issue(s) this PR fixes** 

[THREESCALE-11006](https://issues.redhat.com/browse/THREESCALE-11006)

**Verification steps** 

1. Go to the dashboard and confirm there is a Quick starts button.
2. Click the button and confirm you're redirected to `/p/admin/quickstarts`.
3. Create a new provider.
4. Confirm that on login you receive the Quick starts notification.
5. Confirm that after logout and login again you don't receive the Quick starts notification.

### CODE REVIEW

**Comment by lvillen** [comment]
@dfennessy @lcavalle currently at the dashboard we have a button with string `Quickstarts` that redirect to Quickstarts page which has a header `Quick starts`. Could you please clarify if we should use `Quickstarts` or `Quick starts` to be consistent?

**Comment by dfennessy** [comment]
> @dfennessy @lcavalle currently at the dashboard we have a button with string `Quickstarts` that redirect to Quickstarts page which has a header `Quick starts`. Could you please clarify if we should use `Quickstarts` or `Quick starts` to be consistent?

@lvillen It should be `Quick starts`

File: app/controllers/provider/sessions_controller.rb [hunk 1]
```diff
@@ -22,6 +22,7 @@ def create
 
     if @user
       self.current_user = @user
+      cookies[:first_login] = { value: 'true' } if current_user.user_sessions.empty?
```
Comment (jlledom) [inline app/controllers/provider/sessions_controller.rb:25]: IMO you don't need to use cookies, that's stored on the client and can be manipulated by the user. Specially when the cookie is not encrypted. If any, you should use the store session. However, I don't think you need the session neither. Here you could just store whether this is the first login:

```suggestion
      first = current_user.user_sessions.empty?
```

And then a couple lines below set the flash notice accordingly:

```ruby
flash[:notice] = first ? 'You can use quick starts to learn about 3scale features step by step. <a href="/p/admin/quickstarts">Take tour</a>'.html_safe : 'Signed in successfully'
```
Even better if you use i18n and load the text from `en.yml`

File: app/views/provider/admin/dashboards/show.html.slim [hunk 2]
```diff
@@ -4,6 +4,11 @@
 - content_for :javascripts do
   = javascript_packs_with_chunks_tag 'dashboard'
 
+button class="pf-c-button pf-m-primary pf-c-button__float-button" type="button"
+  span class="pf-c-button__icon pf-m-start"
+    i class="fas fa-info-circle" aria-hidden="true"
+  a href="/p/admin/quickstarts" Quickstarts
```
Comment (jlledom) [inline app/views/provider/admin/dashboards/show.html.slim:10]: I think this is too much and the flash notice should be enough, but if that's what they want...

Comment (josemigallas) [inline app/views/provider/admin/dashboards/show.html.slim:10]:
```suggestion
  a href=provider_admin_quickstarts_path Quickstarts
```

File: test/integration/provider/admin/dashboards_controller_test.rb [hunk 3]
```diff
@@ -29,4 +29,9 @@ class Provider::Admin::DashboardsControllerTest < ActionDispatch::IntegrationTes
     get provider_admin_dashboard_path
     assert_xpath(xpath_selector, element_text)
   end
+
+  test 'receives quickstarts flash on first login and first_login cookie is false' do
+    assert_equal 'You can use quick starts to learn about 3scale features step by step. <a href="/p/admin/quickstarts">Take tour</a>', flash[:quickstarts_notice]
+    assert_equal 'false', cookies[:first_login]
+  end
```
Comment (josemigallas) [inline test/integration/provider/admin/dashboards_controller_test.rb:36]: This is no integration test. I think it's enough with the cucumber.

**Review by josemigallas — CHANGES_REQUESTED** [review]

File: app/controllers/provider/admin/dashboards_controller.rb [hunk 4]
```diff
@@ -16,6 +16,9 @@ def show
     @services           = current_user.accessible_services
     @messages_presenter = current_presenter
     @unread_messages_presenter = unread_messages_presenter
+    return unless flash[:quick_starts_notice]
+
+    flash[:quick_starts_notice] = t('provider.admin.dashboards.quick_starts_html', link: provider_admin_quickstarts_path).html_safe
```
Comment (jlledom) [inline app/controllers/provider/admin/dashboards_controller.rb:21]: In order to make it easier to understand that we are using one flash object to transfer data between controllers and the other to show the notice, I would call the objects `flash[:first_login]` and `flash[:notice]`

```suggestion
    return unless flash[:first_login]

    flash[:notice] = t('provider.admin.dashboards.quick_starts_html', link: provider_admin_quickstarts_path).html_safe
```

**Review by jlledom — APPROVED** [review]

**Pull request merged**